node/src/types/block.rs

```diff
@@ -848,6 +848,18 @@ impl FromBytes for BlockHeader {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct BlockHeaderAndFinalitySignatures {
+    pub block_header: BlockHeader,
+    pub finality_signatures: BTreeMap<PublicKey, Signature>,
+}
+
+impl Display for BlockHeaderAndFinalitySignatures {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
```

Comment:
Outside of EE, our `Display` impls are generally not as verbose as `Debug` impls.  All of the components should have `Display` impls, you just need to decide whether you need to iterate the signatures to display that info, or whether it's reasonable to just print e.g. the length of the map as per `impl Display for BlockSignatures`.

Reply:
Especially the part about the length of entries. This can very easily lead to blowing up the logs into gargantuan size.

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/1033/commits/c3c5dd7449cfde784ed20a5345eecd5817aa2d1c